File: nodejs-wdio/tests/pages/signIn.page.ts
import Page from "./page";
import WebinarsPage from "./webinars.page";
import SignInComponent from "./components/signInForm.comp";
import isSignedIn from "../helpers/signIn.helper";
import { User } from "../types/user";

const SCREEN_SELECTOR = "h2.intro-title"; // Nice seeing you again

class SignInPage extends Page {
  constructor() {
    super(SCREEN_SELECTOR);
  }

  public visit(): void {
    // TODO: refactor to make login dynamic
    super.visit("/#/login");
  }

  public signIn(user: User): typeof WebinarsPage {
    SignInComponent.signIn(user);
    return WebinarsPage;
  }

  public isSignedIn(signIn = true): boolean {
    return isSignedIn(signIn);
  }

  public signInAndReturnFieldsError(user: User): Array<String> {
    return SignInComponent.signInAndReturnFieldsError(user);
  }

  public signInAndReturnFormError(user: User): string {
    return SignInComponent.signInAndReturnFormError(user);
  }
}

export default new SignInPage();

File: ruby-capybara/tests/pages/sign_in_page.rb
require_relative '../pages/base_page'
require_relative '../pages/components/sign_in_form_comp'

SCREEN_SELECTOR = 'h2.intro-title'.freeze

class SignInPage < BasePage
  alias_method :parent_visit, :visit

  def initialize
    super(SCREEN_SELECTOR)
    @sign_in_form = SignInFormComponent.new
  end

  def visit
    parent_visit 'https://app.livestorm.co/#/login'
  end

  def sign_in(user)
    @sign_in_form.sign_in(user)
  end

  def sign_in_and_return_fields_error(user)
    @sign_in_form.sign_in_and_return_fields_error(user)
  end

  def sign_in_and_return_form_error(user)
    @sign_in_form.sign_in_and_return_form_error(user)
  end
end
